redisdump/redisdump.go

```diff
@@ -201,6 +206,23 @@ func getDBIndexes(redisURL string) ([]uint8, error) {
 
 func scanKeys(client radix.Client, db uint8, filter string, keyBatches chan<- []string, progressNotifications chan<- ProgressNotification) error {
 	keyBatchSize := 100
+
+	err := client.Do(radix.Cmd(nil, "SELECT", strconv.Itoa(int(db))))
+	if err != nil {
+		return err
+	}
+
+	if keysMode {
```

Comment:
I think this might be better solved by creating another function that implements the same interface as scanKeys, and picking the right function depending on the parameter. It would also avoid having to set a global variable. 
I need to review also why the DB selection is happening outside this function, it does seem to make more sense here :thinking: